Gordon360/Services/ActivityService.cs

```diff
@@ -308,6 +310,7 @@ public async Task<ACT_INFO> UpdateActivityImageAsync(ACT_INFO involvement, IForm
             // Put current DateTime in filename so the browser knows it's a new file and refreshes cache
             var filename = $"canvasImage_{DateTime.Now:yyyy-MM-dd_HH-mm-ss-fff}.png";
             var imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", involvement.ACT_CDE.Trim(), filename);
+            var url = $"{_serverUtils.GetAddress()}/browseable/uploads/{involvement.ACT_CDE.Trim()}/{filename}";
```

Comment:
Since `ServerUtils.GetAddress` can return `null`, you'll want to handle that case here - probably by throwing an Exception so that we don't partially update the picture when we can't save a valid new URL to the database:

```suggestion
            var serverAddress = _serverUtils.GetAddress();
            if (serverAddress is not string) throw new Exception("Could not upload Involvement Image: Server Address is null");
            var url = $"{serverAddress}/browseable/uploads/{involvement.ACT_CDE.Trim()}/{filename}";
```